code/datums/craft/step.dm

```diff
@@ -117,3 +169,39 @@
 				value = best_value
 				. = I
 
+	else if (reqed_material)
+		for(var/obj/item/I in items)
+			if (istype(I, /obj/item/stack/material))
+				var/obj/item/stack/material/MA = I
+				if (MA.material && (MA.material.name == reqed_material))
+					return I
+
+/datum/craft_step/proc/is_valid_to_consume(var/obj/item/I, var/mob/living/user)
+	var/holder = I.get_holding_mob()
+	//Next we must check if we're actually allowed to submit it
+	if (!holder)
+		//If the item is lying on a turf, it's fine
+		return I
+
+	if (holder != user)
+		//The item is held by someone else, can't use
+		return FALSE
+
+	//If we get here, the item is held by our user
+	if (I.loc != user)
+		//The item must be inside a container on their person, it's fine
+		return I
+
+	//The item is on the user
+	if (user.canUnEquip(I))
+		//We test if they can remove it, this will return false for robot objects
+		return I
+
+
+	if (istype(I, /obj/item/stack))
+		//Robots are allowed to use stacks, since those will only deplete the amount but not destroy the item
+		return I
```

Comment:
That check should go befor `if (user.canUnEquip(I))` line 196

Reply:
sure, that would be slightly more efficient, ill make that change